job-descriptions/software-engineer-backend.md

```diff
@@ -0,0 +1,61 @@
+![logo](https://sourcegraph.com/.assets/img/sourcegraph-light-head-logo.svg)
+
+# Software Engineer - Backend
```

Comment:
@chrismwendt @efritz Do you think this job description covers people who we would want to add to the code intel team? If not, do you have any suggestions on how we could modify this description to be more inclusive, or do you think we should have a separate job description for code intel candidates?

Reply:
Yes, I think this description (backend) fits code intel pretty well, and in particular the phrase "increase the number of people who can understand and write code" closely relates to code intel.

One thought: is code intel important enough to be explicitly listed? Something like: "code search, code intelligence, code review, alerts, and automation".

Reply:
> in particular the phrase "increase the number of people who can understand and write code" closely relates to code intel.

Just FYI, the "About us" section in this job description is the same for all of our job descriptions.

> One thought: is code intel important enough to be explicitly listed? Something like: "code search, code intelligence, code review, alerts, and automation".

Where do you want to explicitly list this? The first sentence in "About the role" does explicitly list automation:

_As a backend engineer at Sourcegraph, you will focus on designing and developing fast and reliable services that power our product (e.g., code search, code intelligence, automation)_

Reply:
Oh, I didn't read the full description 🤦‍♂

Reply:
I'd echo what @efritz said a while back "I think polyglot experience is better than familiarity with any one language especially for this team". Go and TypeScript are important, but a lot of what we've been doing with LSIF indexers has included exploration into various language ecosystems, build systems, and CI. A "wide" skillset is more useful than a "deep" one for code intelligence. That said, all of the skills and nice-to-haves apply to code intelligence, too.

I don't think the differences justify a separate job description for code intel candidates, but perhaps we could add a bullet point or two to the nice-to-haves that are relatively unique for each team:

> - (code intel) Familiarity with many different programming languages and paradigms
> - (code intel) Experience with multiple language ecosystems and build systems

Reply:
@chrismwendt what is the difference between your two suggested points? I think there is some subtlety that I am missing.

Reply:
These points could just as well be merged into one bullet point for the job description. Here's what I had in mind:

By "languages and paradigms", I'm referring to various language constructs: functional (Lisp, Haskell), procedural (C), OOP (Java), GC or no GC, static or dynamic typing, etc.

By "language ecosystems and build systems", I'm referring to package managers and how projects are put together. We wrestle with these when making LSIF indexers.

Reply:
I merged these in 43342524bcdaff4d3663c8255e0e66e664cbb172, PTAL and comment on that part of the new diff if you have further suggestions.